homeassistant/components/motioneye/switch.py

```diff
@@ -0,0 +1,120 @@
+"""Switch platform for motionEye."""
+from __future__ import annotations
+
+from types import MappingProxyType
+from typing import Any, Callable
+
+from motioneye_client.client import MotionEyeClient
+from motioneye_client.const import (
+    KEY_MOTION_DETECTION,
+    KEY_MOVIES,
+    KEY_NAME,
+    KEY_STILL_IMAGES,
+    KEY_TEXT_OVERLAY,
+    KEY_UPLOAD_ENABLED,
+    KEY_VIDEO_STREAMING,
+)
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util import slugify
+
+from . import MotionEyeEntity, listen_for_new_cameras
+from .const import CONF_CLIENT, CONF_COORDINATOR, DOMAIN, TYPE_MOTIONEYE_SWITCH_BASE
+
+MOTIONEYE_SWITCHES = [
+    KEY_MOTION_DETECTION,
+    KEY_TEXT_OVERLAY,
+    KEY_VIDEO_STREAMING,
+    KEY_STILL_IMAGES,
+    KEY_MOVIES,
+    KEY_UPLOAD_ENABLED,
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: Callable
+) -> bool:
+    """Set up motionEye from a config entry."""
+    entry_data = hass.data[DOMAIN][entry.entry_id]
+
+    @callback
+    def camera_add(camera: dict[str, Any]) -> None:
+        """Add a new motionEye camera."""
+        async_add_entities(
+            [
+                MotionEyeSwitch(
+                    entry.entry_id,
+                    camera,
+                    switch_key,
+                    entry_data[CONF_CLIENT],
+                    entry_data[CONF_COORDINATOR],
+                    entry.options,
+                )
+                for switch_key in MOTIONEYE_SWITCHES
+            ]
+        )
+
+    listen_for_new_cameras(hass, entry, camera_add)
+    return True
+
+
+class MotionEyeSwitch(MotionEyeEntity, SwitchEntity):
+    """MotionEyeSwitch switch class."""
+
+    def __init__(
+        self,
+        config_entry_id: str,
+        camera: dict[str, Any],
+        switch_key: str,
+        client: MotionEyeClient,
+        coordinator: DataUpdateCoordinator,
+        options: MappingProxyType[str, str],
+    ) -> None:
+        """Initialize the switch."""
+        self._switch_key = switch_key
+        self._switch_key_friendly_name = " ".join(
+            [w.capitalize() for w in self._switch_key.split("_")]
```

Comment:
I would prefer if we just hardcode the friendly name. 

Also, shouldn't it contain the camera name too?

Reply:
oh nm that's taken care off in the `name` property

Reply:
Changed to hardcoded friendly names.